### PR TITLE
fix python apis that are incompatible with scala apis

### LINKED ISSUE #1141: SpatialBatchNormalization has different Scala and Python initializer
In Scala, val module = SpatialBatchNormalization(nOutput, eps=1e-5, momentum=0.1, affine=true,
                                           initWeight=null, initBias=null, initGradWeight=null, initGradBias=null).
In Python, module = SpatialBatchNormalization(nOutput, eps=1e-5, momentum=0.1, affine=True)

### PR DESCRIPTION
**Pull request by yangw1234**
## What changes were proposed in this pull request?

fix python apis that are incompatible with scala apis

## Related links or issues (optional)
fixed #1141

### CODE REVIEW

**Comment by yangw1234** [comment]
pr validation pass

File: pyspark/bigdl/nn/layer.py [hunk 1]
```diff
@@ -953,12 +951,20 @@ def __init__(self,
                  eps=1e-5,
                  momentum=0.1,
                  affine=True,
+                 init_weight=None,
```
Comment (zhichao-li) [inline pyspark/bigdl/nn/layer.py]: add a test here?

File: pyspark/bigdl/nn/layer.py [hunk 2]
```diff
@@ -1153,12 +1159,20 @@ def __init__(self,
...
+                 init_bias=None,
```
Comment (zhichao-li) [inline pyspark/bigdl/nn/layer.py]: add a test here

File: pyspark/bigdl/nn/layer.py [hunk 3]
```diff
@@ -1153,12 +1159,20 @@ def __init__(self,
...
+                 init_grad_weight=None,
+                 init_grad_bias=None,
                  bigdl_type="float"):
         super(BatchNormalization, self).__init__(None, bigdl_type,
                                                  n_output,
                                                  eps,
                                                  momentum,
-                                                 affine)
+                                                 affine,
+                                                 init_weight,
```
Comment (zhichao-li) [inline pyspark/bigdl/nn/layer.py]: add a test

**Comment by yangw1234** [comment]
pr validation pass

**Comment by yangw1234** [comment]
resolve conflicts and pr validation pass

**Comment by zhichao-li** [comment]
LGTM

**Pull request merged**